java/hiro-client/src/main/java/co/arago/hiro/client/rest/DefaultWebSocketClient.java

```diff
@@ -257,6 +347,62 @@ public synchronized void sendMessage(String message) {
         }
     }
 
+    @Override
+    public synchronized void addEventFilter(Map filter) {
+        String id = getFilterId(filter);
+        String message = getEventRegisterMessage(filter);
+        LOG.log(Level.INFO, "Add filter: " + message);
+        sendMessage(message);
+        eventFilterMessages.put(id, filter);
+    }
+
+    private String getFilterId(Map filter) {
+        String id = String.valueOf(filter.get("filter-id"));
+        if (id == null || StringUtils.isEmpty(id)) {
```

Comment:
Strings.IsNullOrEmpty

Reply:
Strings.isNullOrEmpty is not part of Java 8. I could just use "StringUtils.isEmpty(id)" though, because this handles null values as well. DOH! ;D

Reply:
Just discovered another error here. "String.valueOf(...)" returns the String "null" on a null value. We do NOT want this, but the real null if filter-id is missing. Changed that now.